spec/spec_helper.rb

```diff
@@ -1,18 +1,16 @@
+# Enable legacy providers
+ENV['OPENSSL_CONF'] = File.expand_path(
+  File.join(File.dirname(__FILE__), 'support', 'openssl.conf')
+)
```

Comment:
OpenSSL 3 no longer loads old cyphers by default. This allows the tests to pass. Metasploit-framework will have this file loaded at runtime.

Unfortunately it means direct users of RubySMB will need a similar configuration change. Or we will need to wait for a solution similar to https://github.com/WinRb/rubyntlm/pull/53 to land